WalletWasabi/Wallets/PasswordFinder/PasswordFinderHelper.cs

```diff
@@ -21,7 +21,13 @@ public static class PasswordFinderHelper
 	public static bool TryFind(PasswordFinderOptions passwordFinderOptions, [NotNullWhen(true)] out string? foundPassword, Action<int, TimeSpan>? reportPercentage = null, CancellationToken cancellationToken = default)
 	{
 		foundPassword = null;
-		BitcoinEncryptedSecretNoEC encryptedSecret = passwordFinderOptions.Wallet.KeyManager.EncryptedSecret;
+		BitcoinEncryptedSecretNoEC? encryptedSecret = passwordFinderOptions.Wallet.KeyManager.EncryptedSecret;
+
+		// No secret in the watch-only mode.
+		if (encryptedSecret is null)
+		{
+			return false;
```

Comment:
nop. If we don't have the secret then it is impossible to find the password because we don't have the required input and in that case we should throw an exception. Returning false means failed to find it even.